emcc.py

```diff
@@ -1293,7 +1293,8 @@ def get_last_setting_change(setting):
     if not shared.Settings.WASM_BACKEND and 'EMCC_ALLOW_FASTCOMP' not in os.environ:
       diagnostics.warning('fastcomp', 'the fastomp compiler is deprecated.  Please switch to the upstream llvm backend as soon as possible and open issues if you have trouble doing so')
 
-    if options.no_entry or '_main' not in shared.Settings.EXPORTED_FUNCTIONS:
+    if options.no_entry or ('_main' not in shared.Settings.EXPORTED_FUNCTIONS and
+                            '__start' not in shared.Settings.EXPORTED_FUNCTIONS):
       shared.Settings.EXPECT_MAIN = 0
```

Comment:
this seems confusing as now `EXPECT_MAIN` depends on something that is not `main`.

what do you think about erroring on an export of `_start`? it doesn't seem like that should be manually exported by the user. Or is this normal in other toolchains?

Reply:
EXPECT_MAIN already has the meaning "expect entry point"  or "this is a command and not reactor".     I think we discussed renaming it already but its an internal setting... so I don't think its too important.

Reply:
Perhaps I'll rename this settings to REACTOR and flip it to off by default?   I like that actually.

Reply:
Anyway I think this change can land as is

Reply:
I don't feel strongly, but it feels wrong to me for a user to export `_start` or `wasm_ctors` or other stuff that is automatically exported already, when it's relevant. If the user exports them, it's either redundant or wrong, basically.

With that said, you are much more of a compiler toolchain person than me, so if this feels right to you I'm fine with it.